Sources/Components/Feed/InAppFeedViewModel.swift

```diff
@@ -0,0 +1,312 @@
+        public func refreshFeed(showRefreshIndicator: Bool = false) async {
+            do {
+                if showRefreshIndicator {
+                    DispatchQueue.main.async {
+                        self.showRefreshIndicator = true
+                    }
+                }
+                let originalStatus = feedClientOptions.status
+                let archived: Knock.FeedItemArchivedScope? = feedClientOptions.status == .archived ? .only : nil
+                let status = feedClientOptions.status == .archived ? .all : feedClientOptions.status
+                self.feedClientOptions.archived = archived
+                self.feedClientOptions.status = status
+                self.feedClientOptions.before = nil
+                self.feedClientOptions.after = nil
+                guard let userFeed = try await Knock.shared.feedManager?.getUserFeedContent(options: feedClientOptions) else { return }
+                DispatchQueue.main.async {
+                    self.feed = userFeed
+                    self.feed.pageInfo.before = self.feed.entries.first?.__cursor
+                    self.feedClientOptions.status = originalStatus
+                    if self.showRefreshIndicator {
+                        self.showRefreshIndicator = false
+                    }
+                }
+            } catch {
+                handleFeedError(error)
+            }
+        }
+        
+        /// Fetches a new page of feed items when the user scrolls to the bottom of the feed.
+        public func fetchNewPageOfFeedItems() async {
+            guard let after = self.feed.pageInfo.after else { return }
+            feedClientOptions.after = after
+            Knock.shared.feedManager?.getUserFeedContent(options: feedClientOptions) { result in
+                DispatchQueue.main.async {
+                    switch result {
+                    case .success(let feed):
+                        self.mergeFeedsForNewPageOfFeed(feed: feed)
+                    case .failure(let error):
+                        self.handleFeedError(error)
+                    }
+                }
+            }
+        }
+        
+        /// Determines whether there is another page of content to fetch when paginaating the feedItem list
+        public func isMoreContentAvailable() -> Bool {
+            return feed.pageInfo.after != nil
+        }
+        
+        // MARK: Message Update Methods
+        
+        /// Archives a specific feed item.
+        public func archiveItem(_ item: Knock.FeedItem) async {
+            do {
+                let message = try await Knock.shared.updateMessageStatus(messageId: item.id, status: .archived)
+                
+                // remove local message if update was successful
+                DispatchQueue.main.async {
+                    self.feed.entries = self.feed.entries.filter{ $0.id != message.id }
+                }
```

Comment:
One note here is that the removal of the item from the feed is dependent on what the **filters** are for the feed. So here, we should really only be archiving the items when the `archived` filter is equal to `exclude` (the default). The behavior here should be that we also set the `archived_at` to be the current date, and then **only** exclude when the filter is set to exclude.